### ISSUE TITLE
Expose method to get current key index of an animation

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot-proposals#158]
**Describe the project you are working on:**
Action Game, Card Game

**Describe the problem or limitation you are having in your project:**
I'm trying to get the current key of my animation. Maybe I'm blind, but I've been going over and over both the [Animation resource docs](http://docs.godotengine.org/en/stable/classes/class_animation.html) and the [AnimationPlayer ](https://docs.godotengine.org/en/3.1/classes/class_animationplayer.html)docs and could not find it.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
Many methods of the Animation resource use the key index, but there seems no way to get it.

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
I don't know if people would prefer is at method or property, but I guess it would look something like either
`int keyindex `or
`int animation_track_get_key_index (  )`

**Describe implementation detail for your proposal (in code), if possible:**
I don't speak C++

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I would not know how.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It seems like a reasonable thing to have in core since so many other already existing methods require it.

**Comment by LikeLakers2** [comment]
There is [`Animation.track_find_key`](http://docs.godotengine.org/en/stable/classes/class_animation.html#class-animation-method-track-find-key) -- would this do what you need?

**Comment by golddotasksquestions** [comment]
To be honest I don't quite understand int track_find_key ( int idx, float time, bool exact=false ) const
To get the key index I already need to know the index of the key? how is that supposed to work?

**Comment by LikeLakers2** [comment]
`int idx` within that function refers to the track index, not the key index.

Most of the functions within the Animation class use `int idx` for the track index, and `int key_idx` for the key index. I'll admit it can be a little confusing, and perhaps maybe renaming them would be a good idea.

**Comment by golddotasksquestions** [comment]
Thanks, for the tip, I was not aware idx vs key_idx difference. I agree this should be renamed. Made an issue for the docs [here](https://github.com/godotengine/godot-docs/issues/2856).

**Comment by LikeLakers2** [comment]
@akien-mga or @golddotasksquestions Mind closing this one too, since #32842 was merged?

**Comment by LikeLakers2** [comment]
@golddotasksquestions Would this issue be fixed since https://github.com/godotengine/godot/pull/32842 was merged?

**Comment by golddotasksquestions** [comment]
Not really. I still think a `get_current_key_index()` method in the AnimationPlayer would make things a lot easier, but at least I know a workaround for the time being. 
Thanks to your doc fix others can find this workaround more easily too now.

Given the current resistance to make any additions to the engine, I'd much rather have some of my other proposals considered. For instance:
https://github.com/godotengine/godot-proposals/issues/115
https://github.com/godotengine/godot-proposals/issues/129
https://github.com/godotengine/godot-proposals/issues/144
